### PR TITLE
refactor(rosbag2_to_non_annotated_t4_converter): refactor _convert_image for test

### PR DESCRIPTION
**Pull request by kminoda**
## Description

Refactor self._convert_image() function in rosbag2_to_non_annotated_t4_converter so that the LiDAR-camera synchronization algorithm can be tested with pytest. This PR is expected to be followed by another PR which enables a LiDAR-camera fusion w/ robustness to LiDAR message drops.

## How to review

- Run the test and confirm that it passes
- Run a LiDAR-camera rosbag conversion
- Run a camera-only rosbag conversion

### test data

Any data that could be converted with the current latest tier4_perception_dataset is fine.

### test command

```bash
poetry run pytest tests/utils/test_misc.py
```

## Reference

[TIER IV INTERNAL LINK](https://star4.slack.com/archives/C03S84LDJGG/p1701929878859259)

## Notes for reviewer

None

### CODE REVIEW

File: perception_dataset/rosbag2/rosbag2_to_non_annotated_t4_converter.py [hunk 1]
```diff
@@ -564,90 +562,107 @@ def _convert_image(
...
-                    if generated_frame_index >= self._num_load_frames:
-                        return sample_data_token_list
-
-                    sample_record: SampleRecord = sample_records[generated_frame_index]
-                    sample_token: str = sample_record.token
-                    lidar_unix_timestamp = misc_utils.nusc_timestamp_to_unix_timestamp(
-                        sample_record.timestamp
+                    sample_data_token_list.append(sample_data_token)
+                else:
+                    sample_data_token = self._generate_image_data(
```
[the remark this answers is not in the document]
Comment (kminoda) [reply perception_dataset/rosbag2/rosbag2_to_non_annotated_t4_converter.py:625]: Thank you. Fixed https://github.com/tier4/tier4_perception_dataset/pull/55/commits/80c35fd2288ef8bceb181d3c37f4f3815269fa81

**Comment by miursh** [comment]
@kminoda Could you fix error? https://github.com/tier4/tier4_perception_dataset/actions/runs/7176651398/job/19541860664

**Comment by kminoda** [comment]
@miursh Done!

**Review by miursh — APPROVED** [review]
Thank you!

**Pull request merged**